tsl/test/sql/bgw_policy.sql

```diff
@@ -216,17 +216,79 @@ select check_chunk_oid(:oldest_chunk_id, :reorder_chunk_oid);
 -- Should be noop again
 select * from test_reorder(:reorder_job_id) \gset  reorder_
 
-CREATE TABLE test_table_int(time int);
+CREATE TABLE test_table_int(time bigint, junk int);
 SELECT create_hypertable('test_table_int', 'time', chunk_time_interval => 1);
+create or replace function dummy_now() returns BIGINT LANGUAGE SQL IMMUTABLE as  'SELECT 1::BIGINT';
+create or replace function dummy_now2() returns BIGINT LANGUAGE SQL IMMUTABLE as  'SELECT 2::BIGINT';
 
 CREATE TABLE test_table_perm(time timestamp PRIMARY KEY);
 SELECT create_hypertable('test_table_perm', 'time', chunk_time_interval => 1);
 
 \set ON_ERROR_STOP 0
--- we cannot add a drop_chunks policy on a table whose open dimension is not time
+-- we cannot add a drop_chunks policy on a table whose open dimension is not time and no now_func is set
 select add_drop_chunks_policy('test_table_int', INTERVAL '4 months', true);
 \set ON_ERROR_STOP 1
 
+INSERT INTO test_table_int VALUES (-2, -2), (-1, -1), (0,0), (1, 1), (2, 2), (3, 3);
+
+select set_integer_now_func('test_table_int', 'dummy_now');
```

Comment:
we should add a test that has a permission error on set_integer_now_func if you are not the owner of the hypertable.